update migration docs (#49)
onmax/nimiq-developer-center: migration/migration-technical-details.md

Each phase operates within a specified block window, enabling transactions to be sent to the network at specific points within the PoW chain.

> Our approach consists of targeting a specific block height within the PoW chain designated as the *transition block*. At this block height, the blockchain state will be captured and used to generate the "genesis block" for the PoS chain. Instead of starting from block height 0, both chains will share the same block number. The PoS chain starts once the conditions covered in this document are met.

Reviewer: Instead of starting from block height 0, the PoS chain will start from the same block number as the transition block, if the conditions covered in this document are met

## Activation Phase

Only pre-registered validators can participate in this phase. The activation phase serves the purpose of executing the transition. At the first candidate block's height (block 3 456 000), if at least 80% of the total stake signals readiness for the transition, the candidate block becomes the transition block. A special activation tool captures and migrates the state of the PoW chain at the transition block, including the transaction history, to generate the "genesis block" for the new PoS chain.

The Activation Phase follows a specific sequence of events:
1. The candidate block is mined at block height 3 456 000, which opens the first activation window.
2. The network waits for 10 confirmations to prevent block reversion.

> 3. Once 10 confirmation blocks are mined, registered validators capture the state of block 3 456 000 and start generating the "genesis" block.

Reviewer: 3. After 10 block confirmations, registered validators capture the state at block 3 456 000 and start generating the PoS "genesis" block

> 4. Validators generate the "genesis" block deterministically. The block hash should be the same across all validators.

Reviewer: The resulting block hash should be the same across all validators.

> 4. Validators generate the "genesis" block deterministically. The resulting block hash should be the same across all validators.

Reviewer 2: I'd have it as part of 3:
```suggestion
   Validators generate the "genesis" block deterministically. The resulting block hash should be the same across all validators.
```

> 5. After generating the "genesis" block, validators send their readiness transactions, which include the block hash in the data field.

Reviewer 2: It is a single one:
```suggestion
5. After generating the "genesis" block, validators send their readiness transaction, which includes the block hash in the data field.
```

> 6. The network monitors readiness transactions within a 24 hour window.

Reviewer: The activation tool monitors readiness transactions within a 24 hour window

> 7. If 80% of the total stake (represented by registered validators) sends readiness transactions within 24 hours, the PoS chain starts at block height 3 456 000. If the threshold is not reached, the process automatically moves to the next activation window.

Reviewer: If 80% of the total stake (represented by registered validators) signal they are ready, by sending readiness transactions within 24 hours, the PoS chain starts....

> 7. If 80% of the total stake (represented by registered validators) signals that they are ready, by sending readiness transactions within 24 hours, the PoS chain starts at block height 3 456 000. If the threshold is not reached, the process automatically moves to the next activation window.

Reviewer 2: ```suggestion
7. If 80% of the total stake (represented by registered validators) signals that they are ready, by sending readiness transactions within 24 hours, the PoS chain using the produced genesis block (that has a block height of 3 456 000). If the threshold is not reached, the process automatically moves to the next activation window.
```

> 8. If the 80% readiness threshold is not met within the first 24 hours, a new activation window begins, starting with a new candidate block at the block immediately following 1440 blocks or 24 hours from the first candidate block.

Reviewer 2: ```suggestion
8. If the 80% readiness threshold is not met within the first 24 hours, a new activation window begins, starting with a new candidate block at the block immediately following 1440 blocks or 24 hours from the first candidate block and validators need to send again readiness transactions for the new genesis hash that will be produced with this new transition candidate block.
```

> 9. In subsequent activation windows, validators regenerate the "genesis" block (3 457 440), send readiness transactions again, and the process repeats until 80% readiness is achieved.

Reviewer: In subsequent activation windows, the same exact process is repeated. Selecting a new candidate block that is 24 hours apart from the previous one. Until the PoS chain is started

> 9. In subsequent activation windows, the same exact process is repeated. Select a new candidate block 24 hours apart from the previous one by opening a new activation window until the PoS chain starts

Reviewer 2: ```suggestion
9. There might be subsequent activation windows where the same exact process is repeated. A new candidate block is selected 24 hours apart from the previous one by opening a new activation window until the PoS chain starts (by reaching the 80% thrashold readiness).
```

10. Once 80% readiness is reached, the PoS client starts, and the once candidate block becomes the transition block, marking the start of the Nimiq PoS blockchain.

#### Readiness and Activation Tool

> During the activation phase, validators send a transaction signaling their readiness. Pre-registered validators are encouraged to run the activation tool before the phase begins, as migrating the entire transaction history is time-consuming. An activation tool has been developed to facilitate this process. The tool migrates the history and scans the PoW chain for readiness transactions selecting the valid ones, which come from pre-registered validators who have completed the registration process. Additionally, the tool verifies the data field to ensure the hash generated by the client matches the actual transition block configuration hash.

Reviewer: I did not get the last part " Additionally, the tool verifies the data field to ensure the hash generated by the client matches the actual transition block configuration hash."

Author: the `data` field is verified to check if the hash of the genesis is the same across validators

> During the activation phase, validators send a transaction signaling their readiness. Pre-registered validators are encouraged to run the activation tool before the phase begins, as migrating the entire transaction history is time-consuming. An activation tool has been developed to facilitate this process. The tool migrates the history and scans the PoW chain for readiness transactions selecting the valid ones, which come from pre-registered validators who have completed the registration process. The tool also verifies the `data` field to ensure the hash generated by the client matches the actual transition block configuration hash.

Reviewer 2: ```suggestion
During the activation phase, validators send a transaction signaling their readiness. Pre-registered validators are encouraged to run the activation tool before the phase begins, as migrating the entire transaction history is time-consuming. An activation tool has been developed to facilitate this process. The tool migrates the history and scans the PoW chain for readiness transactions selecting the valid ones, which come from pre-registered validators who have completed the registration process. The tool also verifies the `data` field to ensure the genesis hash that validators signaled readiness for, matches the candidate transition block generated.
```

| Data      | Hash of the generated `GenesisConfig` |
| Validity  | Activation block window               |

#### Transition Block

> The `GenesisConfig` is generated after the candidate block is mined. If the pre-registered validators run the tool before the activation window starts, the tool will have already migrated most of the history by the time the candidate block is mined. After the candidate block is mined and the network reaches 80% readiness, the tool migrates the final part of the history and generates the "genesis" block, which becomes the transition block for the PoS chain.

Reviewer 2: ```suggestion
The `GenesisConfig` is generated after the candidate block is mined. If the pre-registered validators run the tool before the activation window starts, the tool will have already migrated most of the history by the time the candidate block is mined. After the candidate block is mined and the network reaches 80% readiness, the tool migrates the final part of the history and generates the "genesis" block, which becomes the candidate transition block for the PoS chain.
```

> The activation tool activates at the candidate block. At this point, the tool scans and analyzes transactions sent to the network. The transition starts if at least 80% of the total stake is ready.

Reviewer 2: ```suggestion
The activation process activates at the candidate block. At this point, the tool scans and analyzes transactions sent to the PoW chain. The transition starts if at least 80% of the total stake is ready for the specified candidate transition block.
```

> The activation tool activates at the candidate block - the block height aligned with an election block corresponding to the PoS chain’s structure. At this point, the tool scans and analyzes transactions sent to the network. The transition starts if at least 80% of the total stake is ready.

Reviewer: Remove " the block height aligned with an election block corresponding to the PoS chain’s structure"

#### Failing 80% of Readiness

If the transition does not succeed within the initial activation window, a new one opens immediately. Each new window starts at the block following the first 24 hours and spans 1 440 PoW blocks (24 hours), repeating until the 80% readiness target is achieved.

## Nimiq PoS

> Once 80% readiness is achieved, the PoS chain begins. Any subsequent transaction included in the PoW chain after the candidate block is not considered part of the PoS chain. From this point forward, the network operates entirely under the PoS consensus mechanism.

Reviewer 2: ```suggestion
Once 80% readiness is achieved, the PoS chain begins. Any subsequent transaction included in the PoW chain after the candidate transition block is not considered part of the PoS chain. From this point forward, the network operates entirely under the PoS consensus mechanism.
```

> Once 80% readiness is achieved and the final portion of the history is migrated, the transition block is finalized, marking the official beginning of the PoS chain. All PoW blocks mined after the transition block are disregarded as if they never occurred. From this point forward, the network operates entirely under the PoS consensus mechanism.

Reviewer: Once 80% readiness is achieved, the PoS chain begins. Any subsequent transaction that is included in the PoW chain after the candidate block, is not considered as part of the PoS chain.
